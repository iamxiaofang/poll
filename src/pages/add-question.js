import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useNavigate } from "react-router";
import { handleLoadData, handleSaveQuestion } from "../actions";

export const AddQuestionPage = () => {

  const navigate = useNavigate()
  const dispatch = useDispatch();
  const [optionOneText, setOptionOneText] = useState('')
  const [optionTwoText, setOptionTwoText] = useState('')
  const author = useSelector(state => state.login.user.id)
  const handleOptionOne = (e) => {
    setOptionOneText(e.target.value)
  };

  const handleOptionTwo = (e) => {
    setOptionTwoText(e.target.value)
  }

  const handleCreate = (e) => {
    e.preventDefault()
    dispatch(handleSaveQuestion({ optionOneText, optionTwoText, author }))
    dispatch(handleLoadData())
    navigate('/')
  }

  return (
    <form>
      <h1>Would You Rather</h1>
      <p>
        <label>Option one:</label>
        <input value={optionOneText} onChange={handleOptionOne} type="text" />
      </p>
      <p>
        <label>Option two:</label>
        <input value={optionTwoText} onChange={handleOptionTwo} type="text" />
      </p>
      <button onClick={handleCreate}>Create question</button>
    </form>
  )
}