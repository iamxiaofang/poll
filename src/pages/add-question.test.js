import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import configureMockStore from 'redux-mock-store';
import { thunk } from "redux-thunk";
import { AddQuestionPage } from './add-question';

const mockStore = configureMockStore([thunk]);

test('Nav', () => {
  const store = mockStore({
    loading: false,
    questions: {},
    login: {
      user: {
        id: 'testUser',
        name: 'Test User',
        avatarURL: 'https://example.com/avatar.png',
      }
    },
  });

  const component = render(
    <Provider store={store}>
      <MemoryRouter>
        <AddQuestionPage />
      </MemoryRouter>
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
