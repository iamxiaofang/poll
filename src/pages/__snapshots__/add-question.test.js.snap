// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <h1>
          Would You Rather
        </h1>
        <p>
          <label>
            Option one:
          </label>
          <input
            type="text"
            value=""
          />
        </p>
        <p>
          <label>
            Option two:
          </label>
          <input
            type="text"
            value=""
          />
        </p>
        <button>
          Create question
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <h1>
        Would You Rather
      </h1>
      <p>
        <label>
          Option one:
        </label>
        <input
          type="text"
          value=""
        />
      </p>
      <p>
        <label>
          Option two:
        </label>
        <input
          type="text"
          value=""
        />
      </p>
      <button>
        Create question
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
